### PR TITLE
Fix rm permissions

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**

### CODE REVIEW

**Review by fviale — APPROVED** [review]
LGTM

File: common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java [hunk 1]
```diff
@@ -427,19 +430,23 @@ public boolean userHasPermissionToGetRmThreadDump() {
...
+        return sessionPermissions.get(PERMISSION_RM_GET_INFRAS_TO_POLICIES_MAPPING);
+    }
+
+    public boolean userHasPermissionToGetSupportedNodeSourceInfras() {
+        return sessionPermissions.get(PERMISSION_RM_GET_SUPPORTED_NODE_SOURCE_INFRASTRUCTURES);
     }
 
-    public boolean userDoesNotHavePermissionToGetSupportedNodeSourceInfras() {
-        return !sessionPermissions.get(PERMISSION_RM_GET_SUPPORTED_NODE_SOURCE_INFRASTRUCTURES);
+    public boolean userDoesNotHavePermissionToDefineNodeSource() {
```
Comment (AlexFalessi) [inline common-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/common/client/model/LoginModel.java:441]: For the sake of consistency you should do like upper tests and use `userHasPermissionToDefineNodeSource`

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ContextMenu.java [hunk 2]
```diff
@@ -225,20 +225,65 @@ public static Menu createContextMenuFromCompactView(RMController controller, Obj
...
+        boolean permissionForNode = loginModel.userHasProviderPermissionForNode(url);
+        if (!permissionForNode || !loginModel.userHasPermissionToLockNodes()) {
             disableLockMenuItem(menu);
+        }
+        if (!permissionForNode || !loginModel.userHasPermissionToUnLockNodes()) {
             disableUnlockMenuItem(menu);
+        }
+        if (!permissionForNode || !hasMethodPermissionToRemoveItem(menu)) {
             disableRemoveMenuItem(menu);
         }
```
Comment (AlexFalessi) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ContextMenu.java:237]: Isn't testing `!loginModel.userHasPermission...()` and `!hasMethodPermissionToRemoveItem(menu)` methods sufficient ?

If you keep `!permissionForNode` test you can optimize it a little by not testing in each if.
For instance

```
if (!loginModel.userHasProviderPermissionForNode(url)){
	disableLockMenuItem(menu);
	disableUnlockMenuItem(menu);
	disableRemoveMenuItem(menu);
	return;
} else {
	if (!loginModel.userHasPermissionToLockNodes()) {
		disableLockMenuItem(menu);
	} 
	
	if (!loginModel.userHasPermissionToUnLockNodes()) {
		disableUnlockMenuItem(menu);
	}
	
	if (!hasMethodPermissionToRemoveItem(menu)) {
		disableRemoveMenuItem(menu);
	}
}
```

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ContextMenu.java [hunk 3]
```diff
@@ -225,20 +225,65 @@ public static Menu createContextMenuFromCompactView(RMController controller, Obj
...
+        }
+        if (!permissionForNodeSource || !loginModel.userHasPermissionToEditNodeSource()) {
+            disableEditItem(menu);
+        }
+        if (!permissionForNodeSource && !loginModel.userHasPermissionToRemoveNodeSource()) {
             disableRemoveMenuItem(menu);
         }
+        if (!permissionForNodeSource && !hasMethodPermissionToEditItem(menu)) {
+            disableRemoveMenuItem(menu);
+        }
```
Comment (AlexFalessi) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ContextMenu.java:257]: Same comment as above, I think you should optimize the `!permissionForNodeSource` testing

**Review by AlexFalessi — APPROVED** [review]
Small comments. Your call

**Pull request merged**